src/main/java/net/neoforged/neoforge/event/entity/living/DamageBlockEvent.java

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright (c) Forge Development LLC and contributors
+ * SPDX-License-Identifier: LGPL-2.1-only
+ */
+
+package net.neoforged.neoforge.event.entity.living;
+
+import net.minecraft.util.Mth;
+import net.minecraft.world.damagesource.DamageSource;
+import net.minecraft.world.entity.LivingEntity;
+import net.neoforged.bus.api.ICancellableEvent;
+import net.neoforged.neoforge.common.damagesource.DamageContainer;
+
+/**
+ * The ShieldBlockEvent is fired when an entity successfully blocks with a shield.<br>
+ * Cancelling this event will have the same impact as if the shield was not eligible to block.<br>
+ * The damage blocked cannot be set lower than zero or greater than the original value.<br>
+ * Note: The shield item stack "should" be available from {@link LivingEntity#getUseItem()}
+ * at least for players.
+ */
+public class DamageBlockEvent extends DamageSequenceEvent implements ICancellableEvent {
+    private float dmgBlocked;
+    private float shieldDamage = 0;
+    private final boolean originalBlocked;
+    private boolean newBlocked;
+
+    public DamageBlockEvent(LivingEntity blocker, DamageContainer container, boolean originalBlockedState) {
+        super(blocker, container);
+        this.dmgBlocked = container.getOriginalDamage();
+        this.originalBlocked = originalBlockedState;
+        this.newBlocked = originalBlockedState;
+    }
+
+    /**
+     * @return The damage source.
+     */
+    public DamageSource getDamageSource() {
+        return this.getDamageContainer().getSource();
+    }
+
+    /**
+     * @return The original amount of damage blocked, which is the same as the original
+     *         incoming damage value.
+     */
+    public float getOriginalBlockedDamage() {
+        return this.getDamageContainer().getOriginalDamage();
+    }
+
+    /**
+     * @return The current amount of damage blocked, as a result of this event.
+     */
+    public float getBlockedDamage() {
+        return this.dmgBlocked;
+    }
+
+    /**
+     * If the event is {@link #getBlocked()} and the user is holding a shield, the returned amount
+     * will be taken from the item's durability.
+     * 
+     * @return The amount of sheild durability damage to take.
+     */
+    public float shieldDamage() {
+        return this.shieldDamage;
+    }
+
+    /**
+     * Set how much damage is blocked by this action.<br>
+     * Note that initially the blocked amount is the entire attack.<br>
+     */
+    public void setBlockedDamage(float blocked) {
+        this.dmgBlocked = Mth.clamp(blocked, 0, this.getOriginalBlockedDamage());
+    }
+
+    /**
+     * Set if the shield will take durability damage or not.
```

Comment:
This setter doc still reads like a boolean, think you just fixed it on the getter.